PWGCF/FemtoDream/TableProducer/femtoDreamProducerTask.cxx

```diff
@@ -1080,6 +1124,74 @@ struct femtoDreamProducerTask {
     }
   }
 
+  // Separate function for fill flow and qn
+  template <bool isMC, bool useCentrality, bool analysePbPb, typename CollisionType, typename TrackType>
+  void fillCollisionsFlow(CollisionType const& col, TrackType const& tracks)
+  {
+    // If triggering is enabled, select only events which were triggered wit our triggers
+    if (ConfEnableTriggerSelection) {
+      bool zorroSelected = zorro.isSelected(col.template bc_as<aod::BCsWithTimestamps>().globalBC()); /// check if event was selected by triggers of interest
+      if (!zorroSelected) {
+        return;
+      }
+    }
+
+    const auto spher = colCuts.computeSphericity(col, tracks);
+    float mult = 0;
+    int multNtr = 0;
+    if (ConfIsRun3) {
+      if constexpr (useCentrality) {
+        if constexpr (analysePbPb) {
+          mult = col.centFT0C();
+        } else {
+          mult = col.centFT0M();
+        }
+      } else {
+        mult = 0;
+      }
+      multNtr = col.multNTracksPV();
+    } else {
+      mult = 1; // multiplicity percentile is know in Run 2
+      multNtr = col.multTracklets();
+    }
+
+    // check whether the basic event selection criteria are fulfilled
+    // that included checking if there is at least on usable track or V0
+    if (!colCuts.isSelectedCollision(col)) {
+      return;
+    }
+    // bool emptyCollision = false;
+    if (colCuts.isEmptyCollision(col, tracks, trackCuts)) {
+      return;
+    }
+
+    if (rctCut.requireRCTFlagChecker && !rctChecker(col)) {
+      return;
+    }
+
+    // Pileup rejection in PbPb data
+    if constexpr (analysePbPb) {
+      if (OptionEvtSpecialSelections.ConfIsUsePileUp &&
+          !colCuts.isPileUpCollisionPbPb(col, OptionEvtSpecialSelections.ConfEvNoSameBunchPileup, OptionEvtSpecialSelections.ConfEvIsGoodITSLayersAll,
+                                         OptionEvtSpecialSelections.ConfTPCOccupancyMin, OptionEvtSpecialSelections.ConfTPCOccupancyMax)) {
+        return;
+      }
+    }
+
+    // Calculate and fill qn values
+    float myqn = colCuts.computeqnVec(col);
+    outputExtQnCollision(myqn, col.trackOccupancyInTimeRange());
+
+    // Calculate flow via cumulant
+    if (qnCal.ConfFlowCalculate) {
+      int qnBin = colCuts.myqnBin(mult, qnCal.ConfCentralityMax, qnCal.ConfQnBinSeparator, qnCal.ConfdoFillHisto, spher, myqn, qnCal.ConfNumQnBins, multNtr, qnCal.ConfCentBinWidth);
+      if (qnBin < qnCal.ConfQnBinMin || qnBin > qnCal.ConfNumQnBins) {
+        qnBin = -999;
+      }
+      colCuts.doCumulants(col, tracks, mult, qnCal.ConfQnSeparation);
```

Comment:
This is not part of the produced data. Are these cumulants calculated as a QA/needed to see the profile of cumulants to create qn bins later in analysis? (There might not be any problem, I just want to understand.)